### DISCUSSION TITLE
TLS SMTP setup options

### DISCUSSION THREAD

**Discussion by MrRichThomas** [postalserver/postal#2214] [Help with using Postal]
Hi, I've been configuring our Postal V2 server to use TLS for SMTP and wondered if this is the best approach or not. This is how I've set it up:

- Configured Caddy to create RAS4096 certs

- Mounted the caddy-data as a volume in the SMTP > volumes section in docker-compose.yml

 > /opt/postal/caddy-data/caddy/certificates:/certificates

- Changed the permissions on that folder
- Set the postal certs like so:

> tls_certificate_path: /certificates/acme-v02.api.letsencrypt.org-directory/_domainName_/_domainName_.crt
> tls_private_key_path: /certificates/acme-v02.api.letsencrypt.org-directory/_domainName_/_domainName.key_

I've read a few other threads on here where people are copying the certs to the postal config folder and changing the permissions and restarting Postal, all with a cron job.

I think with my setup I'd still need to restart Postal for it to pick up the renewed certs, but I don't need to copy certs or change permissions.

The downside is I need to change the permissions on the Caddy folder, but then I'm not sure if this is any better or worse than copying the certs elsewhere and changing permissions. At the moment I've got the permissions a bit too relaxed, with 'other' having read and execute permissions on that folder recursively. I'm guessing instead I could use the 'setgid' and add docker as the group, would it be the docker group that needs permissions though?

Any thoughts much appreciated!

Thanks

**Comment by willpower232** [comment]
> I think with my setup I'd still need to restart Postal for it to pick up the renewed certs

Unfortunately this is definitely a thing, an unfortunate side effect of using certificates that expire often.

Regarding the permissions, I don't think it would be the docker group, it depends what the user ID of the user is inside the container, I'm not sure what that is off the top of my head.

You could ensure the user ID is present in whatever group Caddy is using but it depends if that overlaps with an existing user in your server.

**Reply by MrRichThomas** [reply]
Thanks for the reply Will, I've decided to use the same approach that you and others have mentioned here and am now copying the certs to the /opt/postal/config folder and setting permissions, all with a weekly cron job. It was easier than messing around with the permissions on the caddy-data folder.

Regarding the restart, I notice there's no 'postal restart' function, but it looks like I can do either 'postal dc restart' or for just the SMTP service I can do 'docker restart postal-smtp-1'. Do you happen to know if the latter is all that's required when the certs have been updated? I'd rather do that than a full postal restart just in case there's any issues.

Also, just out of interest, I notice on other threads about setting up TLS for SMTP that you mention you are using Certbot. Are you running that alongside NGINX? We went with Caddy as the Postal documentation example used it, but we're actually more familiar with NGINX and Certbot as we use them on our web reverse proxy servers. Just wondered what set up you have. Thanks :)

**Reply by willpower232** [reply]
`postal dc restart smtp` is probably the most efficient, yes it is a little annoying there isn't a more direct restart but nothing else uses the certificate anyway.

We have a fair few servers in use all over the place so its easier to centralise our knowledge on nginx but Caddy is certainly not a bad choice and I believe it is only getting better with time.

Using certbot means you can update the cron job which updates the certificates with extra commands to sync the certificates where you need them and reload the services so is a little more convenient.
